packages/ingest/SftpProviderClient.js

```diff
@@ -1,32 +1,33 @@
 'use strict';
 
-const { join } = require('path');
 const { log, sftp: { Sftp } } = require('@cumulus/common');
 const get = require('lodash.get');
 const omit = require('lodash.omit');
 const recursion = require('./recursion');
 
-module.exports.sftpMixin = (superclass) => class extends superclass {
-  constructor(...args) {
-    super(...args);
-    const sftpConfig = {
+class SftpProviderClient {
+  constructor(providerConfig) {
+    this.id = providerConfig.id;
+    this.host = providerConfig.host;
+    this.path = providerConfig.path;
+
+    this.sftpClient = new Sftp({
       host: this.host,
-      port: this.port || 22,
-      username: this.username,
-      password: this.password,
-      encrypted: this.provider.encrypted,
-      privateKey: get(this.provider, 'privateKey', null),
-      cmKeyId: get(this.provider, 'cmKeyId', null)
-    };
-    this.sftpClient = new Sftp(sftpConfig);
+      port: providerConfig.port || 22,
```

Comment:
Same comment as above about validating necessary provider config

Reply:
Not in this case, any missing values will be detected when we create the `Sftp` object in the constructor.